### PR TITLE
Support for datasets

### PR DESCRIPTION
**Pull request by QRemy**
Replace dataset argument by datasets in all methods
Adapt the example given in the README (and checked it runs)

### CODE REVIEW

**Review by dvnll — APPROVED** [review]
Great! Many thanks!

File: ECPLiBase.py [hunk 1]
```diff
@@ -58,29 +58,29 @@ class ECPLiBase(ABC):
...
     """
     def __init__(self,
                  limit_target: LimitTarget,
-                 dataset: Dataset,
+                 datasets: Datasets,
                  CL: float):
 
         self.limit_target = limit_target
-        self.dataset = dataset.copy()
+        self.datasets = datasets.copy()
```
Comment (QRemy) [inline ECPLiBase.py:75]: the copy will be inefficient for large datasets and could likely be avoided as modifications of the original datasets that affect only parameters values/frozen status can be reverted with a context manager as https://docs.gammapy.org/0.18.2/api/gammapy.modeling.Parameters.html#gammapy.modeling.Parameters.restore_values

Comment (dvnll) [reply ECPLiBase.py:75]: I know. Didn't have the time for optimizations.

File: ProfileLikelihoodRatio.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def freeze_target_parameter(frozen: bool) -> None:
             Args:
                 frozen: If true, the target parameter is frozen in the fit.
             """
-            for model in dataset.models:
+            for model in datasets.models:
                 if model.name != self.limit_target.model.name:
```
Comment (QRemy) [inline ProfileLikelihoodRatio.py:84]: could be simplified (here in other places)
`model = datasets.models[self.limit_target.model.name]`

Comment (QRemy) [reply ProfileLikelihoodRatio.py:84]: same for parameters few line below
`parameter = model.parameters[self.limit_target.parameter_name]`

**Pull request merged**